src/lib.rs

```diff
@@ -22,104 +63,192 @@ impl fmt::Display for DeviceType {
     }
 }
 
-#[derive(Serialize, Debug)]
-pub struct Device {
-    pub serial_number: String,
-    pub device_type: DeviceType,
-    pub is_on: bool,
-    pub brightness_in_lumen: u16,
-    pub temperature_in_kelvin: u16,
-    #[serde(skip_serializing)]
-    pub device_handle: HidDevice,
-    pub minimum_brightness_in_lumen: u16,
-    pub maximum_brightness_in_lumen: u16,
-    pub minimum_temperature_in_kelvin: u16,
-    pub maximum_temperature_in_kelvin: u16,
+/// A device-relatred error.
+#[derive(Debug, Clone, PartialEq)]
+pub enum DeviceError {
+    /// Tried to use a device that is not supported.
+    Unsupported,
 }
 
-const VENDOR_ID: u16 = 0x046d;
-const PRODUCT_IDS: [u16; 4] = [0xc900, 0xc901, 0xb901, 0xc903];
-const USAGE_PAGE: u16 = 0xff43;
+/// A device that can be used.
+#[derive(Debug)]
+pub struct Device<'a> {
+    device_info: &'a DeviceInfo,
```

Comment:
No, it's private. This should not be writable, otherwise you might end up with inconsistent objects where type and device info don't match. We should add a method to retrieve the info object though.